src/org/zaproxy/zap/extension/exportreport/Export/ExportReport.java

```diff
@@ -0,0 +1,247 @@
+			this.extension = extension;
+			this.fileExtension = fileExtension;
+		}
+
+		/**
+		 * Main task. Executed in background thread.
+		 */
+		@Override
+		public Void doInBackground() {
+			File f = fc.getSelectedFile();
+
+			String file = f.getName();
+			String absolutePath = f.getAbsolutePath();
+			String fileName = file.substring(0, file.lastIndexOf(fileExtension) - 1);
+			String path = absolutePath.substring(0, absolutePath.lastIndexOf(file));
+
+			String XML_PATH = "";
+			try
+			{
+				XML_PATH = ReportExport.generateDUMP(path, fileName, extension.extensionGetTitle(),
+						extension.extensionGetBy(), extension.extensionGetFor(), extension.extensionGetScanDate(),
+						extension.extensionGetScanVer(), extension.extensionGetReportDate(),
+						extension.extensionGetReportVer(), extension.getDescription(),
+						extension.getIncludedAlertSeverity(), extension.getIncludedAlertDetails());
+			}
+			catch (UnsupportedEncodingException e)
+			{
+				logger.error(e.getMessage(), e);
+				view.showWarningDialog(Constant.messages.getString("exportreport.message.error.dump"));
+			}
+
+			String extensionPath = Constant.getZapHome() + "xml" + File.separator;
+			String mergeXSL = (extensionPath + "merge.xml.xsl");
+			String reportXSL = (extensionPath + "report.html.xsl");
+
+			File f_view = null;
+			boolean show = false;
+			switch (fileExtension.toLowerCase(Locale.ROOT)) {
+			case Utils.html:
+				f_view = ReportExport.TRANSFORMATION(view, logger, absolutePath, (path + fileName + ".xml"), XML_PATH, mergeXSL);
+				f_view = null;
+				f_view = ReportExport.TRANSFORMATION(view, logger, absolutePath, absolutePath,(path + fileName + ".xml"), reportXSL);
+				deleteFile(path + fileName + ".xml", "The merged XML file: ");
+				show = true;
+				break;
+			case Utils.bootstrap:
+				view.showMessageDialog(Constant.messages.getString("exportreport.message.notice.bootstrap"));
+				f_view = null;
+				break;
+			case Utils.xml:
+				f_view = ReportExport.TRANSFORMATION(view, logger, absolutePath, (path + fileName + ".xml"), XML_PATH, mergeXSL);
+				show = true;
+				break;
+			case Utils.json:
+				view.showMessageDialog(Constant.messages.getString("exportreport.message.notice.json"));
+				f_view = ReportExport.TRANSFORMATION(view, logger, absolutePath, (path + fileName + ".xml"), XML_PATH, mergeXSL);
+				f_view = null;
+				f_view = ReportExport.JSON_EXPORT(view, logger, absolutePath, absolutePath,(path + fileName + ".xml"));
+				deleteFile(path + fileName + ".xml", "The merged XML file: ");
+				break;
+			case Utils.pdf:
+				view.showMessageDialog(Constant.messages.getString("exportreport.message.notice.pdf"));
+				f_view = null;
+				break;
+			case Utils.doc:
+				view.showMessageDialog(Constant.messages.getString("exportreport.message.notice.doc"));
+				f_view = null;
+				break;
+			default:
+				break;
+			}
+			deleteFile(XML_PATH, "The data dump file: ");
+			try {
+				if (!(f_view == null) && (show == true))
```

Comment:
Consider using `!=` instead of `!( == )`, also it could just be `show` (without comparison to true).